ftw/tabbedview/statestorage.py

```diff
@@ -32,22 +32,25 @@ def get_key(self):
         key = []
         key.append('ftw.tabbedview')
 
-        # add the portal_type of the context
-        key.append(self.context.portal_type)
+        self._append_portal_type(key)
+        self._append_view_name(key)
+        self._append_userid(key)
 
-        # add the name of the tab
+        # concatenate with "-"
+        return '-'.join(key)
+
+    def _append_view_name(self, key):
         key.append(self.tabview.__name__)
 
+    def _append_portal_type(self, key):
+        key.append(self.context.portal_type)
+
+    def _append_userid(self, key):
         # add the userid
         mtool = getToolByName(self.context, 'portal_membership')
         member = mtool.getAuthenticatedMember()
-        if not member or member == AccessControl.getSecurityManager().getUser():
-            return None
-
-        key.append(member.getId())
-
-        # concatenate with "-"
-        return '-'.join(key)
+        if member and member != AccessControl.getSecurityManager().getUser():
```

Comment:
Why does it raise an `AttributeError`?

The old implementation responded with  `None` to `get_key()` when the user was not authenticated.
The new implementation seems to respond with a key where the member id is missing.

I think it should not be possible to generate an key for anonyous users, since the would share the dictstorage with all other anonymous users. It should not be stored anything at all when the user is anoymous.